### PR TITLE
kvserver/rangefeed: introduce Disconnector interface 

### PR DESCRIPTION
**Pull request by wenyihu6**
This patch introduces the `Disconnector` interface, implemented by the
`registration` interface. Although currently unused, future commits will pass this
interface to `node.MuxRangefeed`, allowing disconnection of individual
registrations at the node level.

Part of: https://github.com/cockroachdb/cockroach/issues/110432
Release note: none

Co-authored-by: Steven Danna danna@cockroachlabs.com

### CODE REVIEW

**Comment by wenyihu6** [comment]
Feel free to hold off on reviews until the previous ones are merged if you prefer.

File: pkg/kv/kvserver/rangefeed/registry.go
Comment (stevendanna) [inline pkg/kv/kvserver/rangefeed/registry.go:28]: [nit] Perhaps

```
Disconnected is a permanent state; once IsDisconnected returns true, it always returns true
```

**Review by stevendanna — APPROVED** [review]

File: pkg/kv/kvserver/rangefeed/registry.go
Comment (wenyihu6) [reply pkg/kv/kvserver/rangefeed/registry.go]: Done.

**Comment by wenyihu6** [comment]
TFTR!

bors r=stevendanna

**Pull request merged**
